examples/api/lib/material/selection_area/selection_area.1.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+/// Flutter code sample for [SelectionArea].
+
+void main() => runApp(const SelectionAreaSelectionListenerExampleApp());
+
+class SelectionAreaSelectionListenerExampleApp extends StatelessWidget {
+  const SelectionAreaSelectionListenerExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  final SelectionListenerNotifier _selectionNotifier = SelectionListenerNotifier();
+  SelectedContentRange? _currentRange;
+  SelectionStatus? _currentSelectionStatus;
+  SelectionListenerStatus? _selectionListenerStatus;
+
+  @override
+  void initState() {
+    super.initState();
+    _selectionNotifier.addStatusListener(_handleOnSelectionStateChanged);
+  }
+
+  @override
+  void dispose() {
+    _selectionNotifier.removeStatusListener(_handleOnSelectionStateChanged);
+    _selectionNotifier.dispose();
+    _currentRange = null;
+    _currentSelectionStatus = null;
+    super.dispose();
+  }
+
+  void _handleOnSelectionStateChanged(SelectionListenerStatus status) {
```

Comment:
Updated implementation to use data interface, and separated the status listener from SelectionListenerNotifier and left it public in SelectableRegionScope. I think this is clearer since the actual notification is coming from the SelectableRegion.